presto-hive/src/main/java/com/facebook/presto/hive/HiveMetadata.java

```diff
@@ -981,6 +982,19 @@ public ConnectorTableHandle createTemporaryTable(ConnectorSession session, List<
         return new HiveTableHandle(schemaName, tableName);
     }
 
+    private Set<ColumnStatisticType> getSupportedColumnStatisticsForTemporaryTable(Type type)
+    {
+        // Temporary table statistics are not committed to metastore, so no need to call metastore for supported
+        // column statistics, instead locally determine.
+        try {
+            return MetastoreUtil.getSupportedColumnStatistics(type);
```

Comment:
since this isn't dependent on the hive supported statistics at all, I would list out supported statistics here rather than calling getSupportedColumnStatistics.